src/core_modules/capture-core/rules/getDataElementsForRulesExecution.js

```diff
@@ -3,8 +3,13 @@ import type { ProgramStage } from '../metaData';
 
 export const getDataElementsForRulesExecution = (stages: Map<string, ProgramStage>) =>
     [...stages.values()]
-        .flatMap(stage => stage.dataElements)
+        .flatMap(stage => stage.stageForm.getElements())
         .reduce((accRulesDataElements, dataElement) => {
-            accRulesDataElements[dataElement.id] = dataElement;
+            accRulesDataElements[dataElement.id] = {
+                id: dataElement.id,
+                valueType: dataElement.type,
+                optionSetId: dataElement.optionSet?.id,
+                name: dataElement.formName || dataElement.name,
+            };
             return accRulesDataElements;
```

Comment:
Should any part of this fix also be applied to the [getDataElementsForRulesExecution](https://github.com/dhis2/capture-app/blob/master/src/core_modules/capture-core/components/WidgetProfile/DataEntry/ProgramRules/getRulesActionsForTEI.js#L39-L51) function in `WidgetProfile`? Besides this doubt, the new approach looks good to me.

Reply:
I had a look at the profile widget now, and it looks fine because it gets the data elements directly from an api response (hence not affected by this refactor of `ProgramStage`).